drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/loaders/project_unit_tracker.rb

```diff
@@ -0,0 +1,69 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# track assignments of units to enrollments / households
+module HmisExternalApis::AcHmis::Importers::Loaders
+  class ProjectUnitTracker
+    attr_reader :assignments, :data_source
+
+    def initialize(data_source)
+      @data_source = data_source
+      projects = Hmis::Hud::Project.where(data_source: data_source)
+
+      # {enrollment_pk => [project_pk, household_id]}
+      @enrollment_lookup = {}
+      # {enrollment_pk => hoh_entry_date}
+      @enrollment_entry_dates = {}
+      enrollment_scope.preload(:project).find_each do |enrollment|
+        @enrollment_lookup[enrollment.id] = [enrollment.project.id, enrollment.household_id]
+        @enrollment_entry_dates[enrollment.id] ||= enrollment.entry_date if enrollment.head_of_household?
+      end
+
+      @household_assignments = {}
+      @assignments = {}
+
+      # we don't check if the unit is occupied, assumption is that all unit occupancies
+      # are deleted before import
```

Comment:
Added to https://docs.google.com/document/d/1TgQ-Y6-49UwQRnpH_7Pz5gnAYDBuzP5ldeS1YL_3z2I/edit